helm/prometheus-rules/templates/platform/atlas/alerting-rules/loki.rules.yml

```
@@ -135,3 +135,22 @@ spec:
         severity: page
         team: atlas
         topic: observability
+  - name: loki.compactor
+    rules:
+    - alert: LokiCompactorFailedCompaction
+      annotations:
+        dashboard: loki-retention/loki-retention
+        description: 'Loki compactor has been failing compactions for more than 2 hours.'
+        opsrecipe: loki#lokicompactorfailedcompaction
+      # This alert checks if Loki's the last successful compaction run is older than 2 hours
+      expr: (time() - loki_compactor_apply_retention_last_successful_run_timestamp_seconds > 60 * 60 * 2)
```

```suggestion
      expr: (time() - loki_boltdb_shipper_compact_tables_operation_last_successful_run_timestamp_seconds > 60 * 60 * 2)
```

According to upstream we should use compaction metric instead of the rétention.